flickit-assessment-data/src/main/java/org/flickit/assessment/data/jpa/kit/kitversion/KitVersionJpaRepository.java

```diff
@@ -17,8 +17,11 @@ public interface KitVersionJpaRepository extends JpaRepository<KitVersionJpaEnti
     @Modifying
     @Query("""
             UPDATE KitVersionJpaEntity k
-            SET k.status = :status
+            SET k.status = :status,
+                k.statusVersion = :statusVersion
             WHERE k.id = :id
         """)
-    void updateStatus(@Param("id") long kitVersionId, @Param("status") int status);
+    void updateStatus(@Param("id") long kitVersionId,
+                      @Param("status") int status,
+                      @Param("statusVersion") long statusVersion);
```

Comment:
**Suggestion:** Ensure that the `updateStatus` method handles the case where `statusVersion` might be incorrectly computed leading to potential data integrity issues. [possible issue, importance: 8]
```suggestion
            @Modifying
            @Query("""
                    UPDATE KitVersionJpaEntity k
                    SET k.status = :status,
                        k.statusVersion = CASE WHEN :status = 0 THEN 0 WHEN :status = 1 THEN 1 ELSE -k.id END
                    WHERE k.id = :id
                """)
            void updateStatus(@Param("id") long kitVersionId,
                              @Param("status") int status,
                              @Param("statusVersion") long statusVersion);
```